### PR TITLE
[ISSUE 7757] Support Persistence Policies on topic level

### PR DESCRIPTION
**Pull request by zhanghaou**
Link [https://github.com/apache/pulsar/issues/7757](https://github.com/apache/pulsar/issues/7757) and master issue [https://github.com/apache/pulsar/issues/2688](https://github.com/apache/pulsar/issues/2688)

### Motivation

Support set/get/remove persistence  policies on topic level.

### Verifying this change
new unit test added.

### CODE REVIEW

**Comment by codelipenghui** [comment]
@jianyun8023 Please help review this PR.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -2294,6 +2295,48 @@ protected void internalGetRetention(AsyncResponse asyncResponse){
...
+            validateGlobalNamespaceOwnership(namespaceName);
+        }
+        Optional<PersistencePolicies> retention = getTopicPolicies(topicName)
+                .map(TopicPolicies::getPersistence);
+        if (!retention.isPresent()) {
+            asyncResponse.resume(Response.noContent().build());
+        }else {
+            asyncResponse.resume(retention.get());
+        }
+    }
```
Comment (jianyun8023) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2311]: minor: there is no need to use `asyncResponse`

Comment (zhanghaou) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: It is same to internalGetRetention(), maybe we can modify it all at once.

**Comment by zhanghaou** [comment]
> The change looks do not handle the topic creation according to the topic policy if the topic level persistent policy exists. You can refer to `createPersistentTopic` method in the BrokerService.java

Done.

**Pull request merged**